api/src/main/java/org/openmrs/validator/VisitValidator.java

```diff
@@ -159,6 +167,50 @@ private void validateStopDatetime(Visit visit, Visit otherVisit, Errors errors)
 			errors.rejectValue("stopDatetime", "Visit.visitCannotContainAnotherVisitOfTheSamePatient", messageBuilder
 			        .toString());
 		}
-		
+
+	}
+
+	private void validateStartDateTimeToBirthdate(Visit visit, Errors errors) {
+
+		Patient patient = visit.getPatient();
+
+		if (patient != null) {
+			if (visit.getStartDatetime() != null && patient.getBirthdate() != null && !patient.getBirthdateEstimated()
+					&& visit.getStartDatetime().before(patient.getBirthdate())) {
+				errors.rejectValue("startDatetime", "Visit.startDateCannotBeBeforeBirthdate",
+						"This visit has a start date before birth date of the patient.");
+			}
+		}
+	}
+
+	private void validateStartDateTimeToEstimatedBirthdate(Visit visit, Errors errors) {
+
+		Patient patient = visit.getPatient();
+
+		if (patient != null) {
```

Comment:
@dkayiwa Are there visits without a patient? Should this not fail with an error?